client/checkout/classic/upe.js

```diff
@@ -556,7 +556,11 @@ jQuery( function ( $ ) {
 				( { error } = await api.getStripe().confirmSetup( upeConfig ) );
 			}
 			if ( error ) {
-				throw error;
+				// Log payment errors on charge and then throw the error.
+				const logError = await api.logPaymentError( error.charge );
+				if ( logError ) {
+					throw error;
+				}
```

Comment:
Does this mean if our logging call fails, we no longer throw the error?

Reply:
Good catch @jrodger, I think we should throw the error irrespective of the log call being successful, we can wrap this promise in a try-catch block for better error handling. That can be done in a separate PR cc @achyuthajoy